fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -256,24 +257,28 @@ public CostEstimate visitPhysicalDistribution(PhysicalDistributionOperator node,
             Preconditions.checkNotNull(statistics);
 
             CostEstimate result;
+            SessionVariable sessionVariable = ConnectContext.get().getSessionVariable();
             DistributionSpec distributionSpec = node.getDistributionSpec();
             switch (distributionSpec.getType()) {
                 case ANY:
                     result = CostEstimate.ofCpu(statistics.getOutputSize(outputColumns));
                     break;
                 case BROADCAST:
-                    if (statistics.getOutputSize(outputColumns) >
-                            ConnectContext.get().getSessionVariable().getMaxExecMemByte()) {
+                    if (statistics.getOutputSize(outputColumns) > sessionVariable.getMaxExecMemByte()) {
                         return CostEstimate.infinite();
                     }
                     int parallelExecInstanceNum = Math.max(1, getParallelExecInstanceNum(context));
+                    int pipelineDop = sessionVariable.isEnablePipelineEngine() && sessionVariable.getPipelineDop() > 0 ?
+                            sessionVariable.getPipelineDop() : 1;
                     // beNum is the number of right table should broadcast, now use alive backends
                     int beNum = Math.max(1, Catalog.getCurrentSystemInfo().getBackendIds(true).size());
                     result = CostEstimate
                             .of(statistics.getOutputSize(outputColumns) *
                                             Catalog.getCurrentSystemInfo().getBackendIds(true).size(),
-                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum,
-                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum);
+                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum *
+                                            pipelineDop,
```

Comment:
pipelineDop is 0 means that the query executed in pipeline engine use the half number of cpu cores. so here suggest to recifty pipelineDop here to a reasonable value when it is 0. 
i.e.:
`beNum * parallelExecInstanceNum * pipelineDop<=0?32:pipelineDop`

In future, the number of cores of BEs should be perceived by FE.

Reply:
Done